src/main/kotlin/no/nav/syfo/sykmeldingstatus/kafka/KafkaFactory.kt

```diff
@@ -7,13 +7,15 @@ import no.nav.syfo.model.sykmeldingstatus.SykmeldingStatusKafkaMessageDTO
 import no.nav.syfo.sykmeldingstatus.kafka.producer.SykmeldingStatusKafkaProducer
 import no.nav.syfo.sykmeldingstatus.kafka.util.JacksonKafkaSerializer
 import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerConfig
 
 class KafkaFactory private constructor() {
     companion object {
         fun getSykmeldingStatusKafkaProducer(kafkaBaseConfig: Properties, environment: Environment): SykmeldingStatusKafkaProducer {
             val kafkaStatusProducerConfig = kafkaBaseConfig.toProducerConfig(
                     "${environment.applicationName}-producer", JacksonKafkaSerializer::class
             )
+            kafkaStatusProducerConfig[ProducerConfig.RETRIES_CONFIG] = 50
```

Comment:
Hvordan vil dette fungere i praksis? Denne skrivingen til topic trigges av en synkron request (i dag kommer den fra syfoservice, senere vil den komme fra frontend). Hvis den skal prøve 50 ganger vil vel den synkrone forbindelsen ha timet ut, brukeren har fått beskjed, og antageligvis forsøkt på nytt. Vil vi ikke potensielt få en del duplikater hvis disse meldingene går igjennom til slutt? 

Og, vil ikke denne retryen holde på tråden til den er ferdig, slik at appen kan slippe opp for tråder hvis det er et midlertidig problem med skriving til kafka..?

Reply:
Denne bruker max 5 sekunder på retries da det er 50 retries (derfor jeg setter den ned fra det som er i common). siden dette trigges av et synkront call som du sier så vil ikke requester returnerer før den enten er OK eller feiler.  

Så om det er mange brukere som forsøker å sette status så vil de bli køet opp og eventuelt time ut, og det er jo forsåvidt greit da det skal feile når vi ikke får skrevet til kafka.

Reply:
Ah, du har satt den NED ja! 😅  Trodde du hadde økt den jeg. 

Men, det er mye bruker kan gjøre som ikke innebærer skriving til kafka, så jeg mener fortsatt den er for høy. Denne appen skal jo brukes for å vise sykmeldinger, og vi har mange flere kall til "hent sykmelding" enn for å sende/bekrefte/endre status på sykmelding. Sånn sett mener jeg det er uheldig hvis appen brenner alle connections på å vente på skriving til kafka slik at den ikke lenger kan hente sykmeldinger for andre brukere. Da synes iallefall jeg at det er bedre at kallet feiler med en gang og at bruker må prøve en gang til :)

Reply:
kan godt sette den ned litt mer, kanskje 2-5 retries?

Reply:
Vi kan jo starte der iallefall?